pkg/server/router.go

```diff
@@ -3,17 +3,20 @@ package server
 import (
 	"net/http"
 
+	"github.com/didip/tollbooth"
+	"github.com/didip/tollbooth_gin"
 	"github.com/gin-contrib/cors"
 	"github.com/gin-gonic/gin"
 )
 
 // SetupRouter sets up the router
 func (s *Server) SetupRouter() *gin.Engine {
 	r := gin.Default()
+	limiter := tollbooth.NewLimiter(s.MaxPerSecond, nil).SetIPLookups([]string{"X-Real-IP", "RemoteAddr", "X-Forwarded-For"})
 	r.Use(cors.Default())
 
 	// Hello there
-	r.GET("/", func(c *gin.Context) {
+	r.GET("/", tollbooth_gin.LimitHandler(limiter), func(c *gin.Context) {
```

Comment:
Do we really RL on that ? ^^

Reply:
not really, thought it would be funny if someone got rate limited trying to refresh the art too much